File: service/src/main/java/net/huray/backend/minuting/service/MinutesWriteService.java
package net.huray.backend.minuting.service;

import lombok.RequiredArgsConstructor;
import net.huray.backend.minuting.dto.MinutesDto;
import net.huray.backend.minuting.entity.MinutesEntity;
import net.huray.backend.minuting.repository.MinutesRepository;
import org.modelmapper.ModelMapper;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.server.ResponseStatusException;

@Service
@RequiredArgsConstructor
@Transactional
public class MinutesWriteService {

    private final MinutesRepository minutesRepository;
    private final ModelMapper modelMapper;

    public MinutesDto.MinutesSimple create(MinutesDto.CreateReq req) {
        MinutesEntity minutesEntity = minutesRepository.save(new MinutesEntity(req.getTitle(), req.getContents()));

        return modelMapper.map(minutesEntity, MinutesDto.MinutesSimple.class);
    }

    public void update(Long id, MinutesDto.UpdateReq req) {
        MinutesEntity minutesEntity = minutesRepository.findById(id)
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));

        minutesEntity.updateMinutes(req.getTitle(), req.getContents());
    }

    public void hardDelete(Long id) {
        minutesRepository.deleteById(id);
    }
}

File: service/src/main/kotlin/net/huray/backend/minuting/service/MinutesWriteService.kt
package net.huray.backend.minuting.service

import net.huray.backend.minuting.dto.MinutesDto
import net.huray.backend.minuting.entity.MinutesEntity
import net.huray.backend.minuting.repository.MinutesRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class MinutesWriteService(
    private val minutesRepository: MinutesRepository
) {

    fun create(req: MinutesDto.CreateReq) = minutesRepository.save(MinutesEntity(req.title, req.contents))
        .let { MinutesDto.MinutesSimple(it.id, it.title, it.contents) }

    fun update(id: Long, req: MinutesDto.UpdateReq) = minutesRepository.findById(id)
        .orElseThrow { throw Exception() }
        .run { updateMinutes(req.title, req.contents) }

    fun hardDelete(id: Long) = minutesRepository.deleteById(id)

}